protocol/credentials.go

```diff
@@ -130,3 +133,21 @@ type CredentialStatusUpdateMessageBody struct {
 	ID     string `json:"id"`
 	Reason string `json:"reason"`
 }
+
+// CredentialRefreshMessage represent Iden3message for credential refresh message
+type CredentialRefreshMessage struct {
+	ID       string                    `json:"id"`
+	Typ      iden3comm.MediaType       `json:"typ,omitempty"`
+	Type     iden3comm.ProtocolMessage `json:"type"`
+	ThreadID string                    `json:"thid,omitempty"`
+
+	Body CredentialRefreshMessageBody `json:"body,omitempty"`
+
+	From string `json:"from,omitempty"`
+	To   string `json:"to,omitempty"`
+}
+
+// CredentialRefreshMessageBody is msg body for refresh message
+type CredentialRefreshMessageBody struct {
+	ID string `json:"id"`
```

Comment:
maybe it should be an array of ID's like an credential offer, and reason field might be applicable

Reply:
Good point. In case we have multi proof request and all claims are revoked we need to refresh the scope of creds.

Reply:
I think that the **reason** field is redundant. What should the user submit in it? We already know that the only case where the refresh service can receive this message is a reissue the credential.

Reply:
it is always useful to let leave a place for explanation.